### ISSUE TITLE
gbtmaker status - error gbtmaker init failure

### ISSUE THREAD

**Issue by yurisyb** [btccom/btcpool#289]
my config.cfg is:

gbtmaker = {
  rpcinterval = 10;
  is_check_zmq = true;
  rawgbt_topic = "BtcRawGbt";
  lightgbt = false; # if unspecified, default false
};
bitcoind = {
  zmq_addr = "tcp://127.0.0.1:8331";
  rpc_addr    = "http://127.0.0.1:8332";
  rpc_userpwd = "bitcoinrpc:usertest123";  # username:password
};
kafka = {
  brokers = "10.1.2.33:9092"; # "10.0.0.1:9092,10.0.0.2:9092,..."
};

error log:

Running on machine: testpool
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
 Utils.cc:272] server responded with code: 401
 BitcoinUtils.cc:394] rpc getnetworkinfo call failure
 Utils.cc:272] server responded with code: 401
 BitcoinUtils.cc:429] rpc getinfo call failure
 GbtMakerMain.cc:139] gbtmaker init failure

fatal error:
gbtmaker init failure

info error:
Running on machine: testpool
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
 GbtMakerMain.cc:98] LTC gbtmaker version 2.3.2-291-g2cb2649
 Kafka.cc:368] producer librdkafka version: 0.9.1
 Kafka.cc:442] add brokers: 10.1.2.33:9092
 Kafka.cc:449] create topic handle: BtcRawGbt
 Utils.cc:272] server responded with code: 401
 BitcoinUtils.cc:394] rpc getnetworkinfo call failure
 Utils.cc:272] server responded with code: 401
 BitcoinUtils.cc:429] rpc getinfo call failure
 GbtMakerMain.cc:139] gbtmaker init failure

Note: I have changed the Btc to Ltc and still getting the same error,

Thanks

**Comment by YihaoPeng** [comment]
HTTP status 401: Unauthorized
Wrong username or password.

Change your bitcoind.conf like this:
```
rpcuser=bitcoinrpc
rpcpassword=usertest123

# add your ip to here if you got a 403 code
rpcallowip=172.16.0.0/12
rpcallowip=192.168.0.0/16
rpcallowip=10.0.0.0/8
```

**Comment by yurisyb** [comment]
i have changed the above configuration and getting below error:

error:
 threadid file:line] msg
14484 Utils.cc:260] unable to request data from: http://127.0.0.1:8332, error: Couldn't connect to server
14484 BitcoinUtils.cc:394] rpc getnetworkinfo call failure
14484 Utils.cc:260] unable to request data from: http://127.0.0.1:8332, error: Couldn't connect to server
14484 BitcoinUtils.cc:429] rpc getinfo call failure
 14484 GbtMakerMain.cc:139] gbtmaker init failure

fatal.error:
gbtmaker init failure

info.error:
14850 GbtMakerMain.cc:98] LTC gbtmaker version 2.3.2-291-g2cb2649
Kafka.cc:368] producer librdkafka version: 0.9.1
Kafka.cc:442] add brokers: 10.1.2.33:9092
Kafka.cc:449] create topic handle: BtcRawGbt
Utils.cc:260] unable to request data from: http://127.0.0.1:8332, error: Couldn't connect to server
BitcoinUtils.cc:394] rpc getnetworkinfo call failure
Utils.cc:260] unable to request data from: http://127.0.0.1:8332, error: Couldn't connect to server
BitcoinUtils.cc:429] rpc getinfo call failure
GbtMakerMain.cc:139] gbtmaker init failure

I changed the 127.0.0.1 to local ip and public ip still getting the same error.

thanks,

**Comment by YihaoPeng** [comment]
You can try connect your bitcoind via curl:
```
curl -v http://127.0.0.1:8332
```
I think you will get the same error.

```
netstat -lntp
```
It should show that no process is listening on port 8332.

This means that your bitcoind does not properly start RPC port listening.

Maybe you need the following configuration:

```
server=1
rpcbind=0.0.0.0
rpcport=8332
rpcuser=bitcoinrpc
rpcpassword=usertest123
rpcallowip=0.0.0.0/0
```

Note: `rpcallowip=0.0.0.0/0` is not unsafe for production environment.

**Comment by yurisyb** [comment]
When i try this commend:
curl -v http://127.0.0.1:8332
**error:**
root@testpool:/work/bitcoind# curl -v http://127.0.0.1:8332
* Rebuilt URL to: http://127.0.0.1:8332/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* connect to 127.0.0.1 port 8332 failed: Connection refused
* Failed to connect to 127.0.0.1 port 8332: Connection refused
* Closing connection 0
curl: (7) Failed to connect to 127.0.0.1 port 8332: Connection refused

It should show that no process is listening on port 8332.
Ans: yes, there is no listening on port 8332.

Where to apply this code? /work/bitcoind/bitcoind.conf or another place
server=1
rpcbind=0.0.0.0
rpcport=8332
rpcuser=bitcoinrpc
rpcpassword=usertest123
rpcallowip=0.0.0.0/0

**Comment by yurisyb** [comment]
When i try this commend:
curl -v http://127.0.0.1:8332

* Rebuilt URL to: http://127.0.0.1:8332/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8332 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8332
> User-Agent: curl/7.58.0
> Accept: */*
>
< HTTP/1.1 405 Method Not Allowed
< Date: Mon, 10 Jun 2019 07:28:20 GMT
< Content-Length: 41
< Content-Type: text/html; charset=ISO-8859-1
<
* Connection #0 to host 127.0.0.1 left intact
JSONRPC server handles only POST requestsroot@testpool:/work/bitcoind#


netstat -lntp
tcp        0      0 127.0.0.1:8332          0.0.0.0:*               LISTEN      16688/bitcoind

fatal error:

Running on machine: testpool
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0610 15:32:04.378659  7897 GbtMakerMain.cc:98] LTC gbtmaker version 2.3.2-291-g2cb2649
I0610 15:32:04.379009  7897 Kafka.cc:368] producer librdkafka version: 0.9.1
I0610 15:32:04.379216  7897 Kafka.cc:442] add brokers: 10.1.2.33:9092
I0610 15:32:04.379272  7897 Kafka.cc:449] create topic handle: LtcRawGbt
E0610 15:32:04.661846  7897 Utils.cc:272] server responded with code: 401
E0610 15:32:04.662313  7897 BitcoinUtils.cc:394] rpc getnetworkinfo call failure
E0610 15:32:04.912911  7897 Utils.cc:272] server responded with code: 401
E0610 15:32:04.912956  7897 BitcoinUtils.cc:429] rpc getinfo call failure
F0610 15:32:04.912984  7897 GbtMakerMain.cc:139] gbtmaker init failure

**Comment by YihaoPeng** [comment]
Is your `bitcoind` and `sserver` running in the same system? If not, you should not use `127.0.0.1`.

And you need keep the `bitcoind` running when you start `gbtmaker`. `bitcoind` is not a background process so if you press `ctrl+c` it will exit.

401 means wrong username or password, use curl to test it:
`curl -v --user bitcoinrpc:mR1MCBgueli5qMcMSQckv9MF6Kk3oD --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "getblocktemplate", "params": [] }' -H 'content-type: text/plain' http://localhost:8332`

If you got a 401 error too, please change the password in your `bitcoin.conf` (not the `gbtmaker.cfg`, it just a client that likes your `curl`).

**Comment by YihaoPeng** [comment]
A Notice: You built the Litecoin version of BTCPool, you can't mine Bitcoin.

**Comment by YihaoPeng** [comment]
No new progress, temporarily closed.
